### PR TITLE
Fixes/Test for External Site Redirect

### PR DESCRIPTION
**Pull request by sebworks**
Fixes/Test for External Site Redirect

## Additions

- Added Unit test for External Site Redirect.

## Removals

- Removed the need for jQuery.

## Changes

- Changed the regex to fix an issue where it wasn't catching query-string params. 

## Testing

- `Visit http://localhost:8000/` and click links in the footer. All non gov links should be redirected.

## Review

- @jimmynotjim 
- @KimberlyMunoz 
- @anselmbradford 

## Notes

- This assumes there is only one level nested spans within an anchor element. 
- The redirect test is very expensive to run on every click. In a beautiful world this wouldn't exist.

## Checklist

* [x] Changes are limited to a single goal (no scope creep)
* [x] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [x] Passes all existing automated tests
* [x] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

File: cfgov/unprocessed/js/modules/external-site-redirect.js [hunk 1]
```diff
@@ -6,29 +6,39 @@
...
-        ( /(https?:\/\/(?:www\.)?(?!.*gov)(?!(content\.)?localhost).*)/g )
-        .exec( url );
+  for ( var i = 0; i < elements.length; ++i ) {
+    elements[i].addEventListener( 'click', _handleClick, false );
+  }
+}
+
+/**
+ * Handle the element click event.
+ * @param {ClickEvent} event A click event.
```
Comment (anselmbradford) [inline cfgov/unprocessed/js/modules/external-site-redirect.js]: Trivial: Type is `{MouseEvent}`

File: cfgov/unprocessed/js/modules/external-site-redirect.js [hunk 2]
```diff
@@ -6,29 +6,39 @@
...
+ */
+function _handleClick( event ) {
+  var element = event.target;
+  if ( element && element.tagName !== 'A' ) element = element.parentNode;
+  if ( element === null || element.tagName !== 'A' ) return;
```
Comment (anselmbradford) [inline cfgov/unprocessed/js/modules/external-site-redirect.js]: Looks like these two lines could be simplified to:

```
if ( element === null || element.tagName !== 'A' ) return;
element = element.parentNode;
```

Comment (sebworks) [inline cfgov/unprocessed/js/modules/external-site-redirect.js]: So the problem is that certain anchor elements contain spans. The event.target will be a span in that scenario.
`<a><span>text</span>`

Comment (anselmbradford) [inline cfgov/unprocessed/js/modules/external-site-redirect.js]: What does `event.currentTarget` give you in that case?

Comment (sebworks) [inline cfgov/unprocessed/js/modules/external-site-redirect.js]: I'm pretty sure it will give you `#main` or `footer` per https://developer.mozilla.org/en-US/docs/Web/API/Event/currentTarget.

File: test/unit_tests/modules/external-site-redirect-spec.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+var jsdom = require( 'mocha-jsdom' );
+
+var clickEvent;
+var link;
+var span;
+var shouldRedirectURLArray = [
+  'http://www.consumerfinance.net/',
+  'http://www.google.com/',
+  'https://www.facebook.com/dialog/share?app_id=210516218981921' +
+  '&display=page&href=http%3A//test.demo.gov/' +
```
Comment (KimberlyMunoz) [inline test/unit_tests/modules/external-site-redirect-spec.js]: This is a great URL to test.

File: test/unit_tests/modules/external-site-redirect-spec.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+    it( 'should handle click events on anchor children', function() {
+      shouldNotRedirectURLArray.forEach( function( url ) {
+        link.href = url;
+        span.dispatchEvent( clickEvent );
+        expect( window.location.href ).to.equal( SITE_URL );
+      } );
+    } );
+
+    it( 'should do nothing when a click event is fired on non achor elements',
```
Comment (KimberlyMunoz) [inline test/unit_tests/modules/external-site-redirect-spec.js]: `non-anchor` typo

File: test/unit_tests/modules/external-site-redirect-spec.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+    it( 'should not redirect internal links', function() {
+      shouldNotRedirectURLArray.forEach( function( url ) {
+        link.href = url;
+        link.dispatchEvent( clickEvent );
+        expect( window.location.href ).to.equal( SITE_URL );
+      } );
+    } );
+
+    it( 'should handle click events on anchor children', function() {
```
Comment (KimberlyMunoz) [inline test/unit_tests/modules/external-site-redirect-spec.js]: I found this one to be difficult to understand, especially if it fails without the context of the other tests. "It should handle click events when anchor links have `spans`"

**Comment by KimberlyMunoz** [comment]
:+1: 

A few minor suggestions, but it looks great. Good job on those edge-cases!

File: test/unit_tests/modules/external-site-redirect-spec.js (hunk 5 above)
Comment (sebworks) [inline test/unit_tests/modules/external-site-redirect-spec.js]: good point, will make the change.

File: cfgov/unprocessed/js/modules/external-site-redirect.js (hunk 2 above)
Comment (sebworks) [inline cfgov/unprocessed/js/modules/external-site-redirect.js]: yeah, it gives you `#main` or `footer`.

File: test/unit_tests/modules/external-site-redirect-spec.js (hunk 5 above)
Comment (sebworks) [inline test/unit_tests/modules/external-site-redirect-spec.js]: Fixed in https://github.com/cfpb/cfgov-refresh/commit/41659c81cfa1ca377f34054370d6ee9767480c09

File: test/unit_tests/modules/external-site-redirect-spec.js (hunk 4 above)
Comment (sebworks) [inline test/unit_tests/modules/external-site-redirect-spec.js]: Fixed in https://github.com/cfpb/cfgov-refresh/commit/41659c81cfa1ca377f34054370d6ee9767480c09

File: test/unit_tests/modules/external-site-redirect-spec.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+      shouldNotRedirectURLArray.forEach( function( url ) {
+        link.href = url;
+        span.dispatchEvent( clickEvent );
+        expect( window.location.href ).to.equal( SITE_URL );
+      } );
+    } );
+
+    it( 'should handle click events when anchor links have spans', function() {
```
Comment (KimberlyMunoz) [inline test/unit_tests/modules/external-site-redirect-spec.js]: Isn't this testing on non-anchor links?

Comment (sebworks) [inline test/unit_tests/modules/external-site-redirect-spec.js]: lol, I'm so blind. Fixed in https://github.com/cfpb/cfgov-refresh/commit/dad321721c7d7431ee10fd05e0872c5db25f891c

Comment (KimberlyMunoz) [inline test/unit_tests/modules/external-site-redirect-spec.js]: :+1:

**Comment by jimmynotjim** [comment]
:+1: Works locally and tests pass
